Documentation/app-operations.md

```diff
@@ -0,0 +1,225 @@
+# Fine-Grained Pod Manipulations
+To provide imperative application (application as in application container and the pod being the OS container) 
+operations (e.g. start|stop|inject) capabilities for finer-grained control over rkt containerization concepts 
+and debugging needs, this proposal introduces new stage1 entrypoints that will be used for manipulating applications inside pods.
+
+The motivation behind this change is the direction orchestration systems are taking
+and to facilitate this new direction. For more details, see [kubernetes#25899](https://github.com/yujuhong/kubernetes/blob/08dc66113399c89e31f6872f3c638695a6ec6a8d/docs/proposals/container-runtime-interface-v1.md).
+
+# API
+
+The envisioned workflow for the app-level CLI API is that After a pod has been started, 
+the users will be invoking the rkt CLI which is application logic on top of stage1 entrypoints (introduced below).
+
+## rkt app inject
+Injects an application image into a running pod.
+
+It first prepares an application rootfs in stage0 for the application image/images
+and then the prepared-app is injected via the entrypoint.
+
+```bash
+rkt app inject <pod-uuid> <image-name/hash/address/registry-URL> <arguments>
+```
+
+## rkt app eject
+Ejects an application image into a running pod, meaning that the leftover resources are removed from the pod.
+
+```bash
+rkt app eject <pod-uuid> --app=<app-name> <arguments>
+```
+
+### Resources Leftover by an Application in Default Stage1 Flavor
+- Rootfs (e.g. `/opt/stage/<app-name>`)
+- Mounts from volumes (e.g. `/opt/stage/<app-name>/<volume-name>`)
+- Mounts related to rkt operations (e.g. `/opt/stage/<app-name>/dev/null`)
+- Systemd service files (e.g. `<app-name>.service` and `reaper-<app-name>.service`)
+- Miscellaneous files (e.g. `/rkt/<app-name>.env`, `/rkt/status...`)
+
+## rkt app start
+Starts an application that was either stopped or injected.
+This operation is idempotent.
+
+```bash
+rkt app start <pod-uuid> --app=<app-name> <arguments>
+```
+
+## rkt app stop
+Stops a running application gracefully (grace is defined in the `app stop` entrypoint section).
+
+```bash
+rkt app stop <pod-uuid> --app=<app-name> <arguments>
+```
+
+## rkt app list
```

Comment:
Its not specified at the moment and its up for discussion, but an app ID (app name is fine since its unique inside a pod atm) and status is the minimal requirement here.